shared_model/backend/protobuf/common_objects/amount.hpp

```diff
@@ -58,10 +58,16 @@ namespace shared_model {
         ValueType &value,
         const boost::multiprecision::uint256_t &amount) noexcept {
       constexpr auto offset = 64u;
-      value.set_first((amount >> offset * 3).template convert_to<uint64_t>());
-      value.set_second((amount >> offset * 2).template convert_to<uint64_t>());
-      value.set_third((amount >> offset).template convert_to<uint64_t>());
-      value.set_fourth(amount.template convert_to<uint64_t>());
+      constexpr boost::multiprecision::uint256_t mask_bits =
+          std::numeric_limits<uint64_t>::max();
+      auto convert = [&](auto i) {
+        return (amount >> offset * i & mask_bits)
```

Comment:
Please add parenthesis to make order of operations more clear.

Reply:
Also maybe a comment explaining what's going on and why we need to zero out unused bytes before cast